assets/app/view/game/buy_sell_shares.rb

```diff
@@ -75,6 +76,20 @@ def render_ipo_shares
         end
       end
 
+      def render_corporate_shares
+        @corporation.corporate_shares.group_by(&:corporation).values.flat_map do |corp_shares|
```

Comment:
won't this have nils in it?

Reply:
I don't think so as @corporation.corporate_shares will return an empty array if there are no shares held by corporations. The group_by and values calls then cannot return nils.

Could you be more specific about what "this" is that might have nils?

Reply:
next unless @step.can_buy?(@current_entity, share)

Reply:
So, the short answer is yes. However, so do the other share render (render_ipo_shares, render_market_shares) functions currently, as the results all get compacted in the render function.